File: src/1463.py
# 1463. 1로 만들기
# 20181229

n=int(input())

dp=[0, 0, 1, 1]

for i in range(4, n + 1):
    dp.append(dp[i - 1] + 1)
    if i % 2 == 0:
        dp[i] = min(dp[i - 1], dp[i//2]) + 1
    if i % 3 == 0:
        dp[i] = min(dp[i - 1], dp[i//3]) + 1

print(dp[n])

File: src/1463.cpp
#include <ios>
#define MIN(a, b) (((a) < (b)) ? (a) : (b))
int main() {
    int N;
    scanf("%d", &N);
    int dp[1000001] = { 0, 0, 1, 1, };
    for (int i=4; i<=N; i++) {
        dp[i] = dp[i-1] + 1;
        if (i % 3 == 0)
            dp[i] = MIN(dp[i], dp[i/3] + 1);
        if (i % 2 == 0)
            dp[i] = MIN(dp[i], dp[i/2] + 1);
    }
    printf("%d", dp[N]);
    return 0;  
}
